vw/src/main/scala/com/eharmony/spotz/objective/vw/VwFunctions.scala

```diff
@@ -0,0 +1,157 @@
+          case it: Iterable[_] => s"$dashes$vwArg ${it.mkString}"
+          case x => s"$dashes$vwArg ${x.toString}"
+        }.mkString(" ") + " "
+        case _ => s"$dashes$vwArg $vwValue "
+      }
+      sb ++= vwParam
+    }.toString()
+  }
+
+  def getTrainVwParams(vwParamMap: Map[String, _], point: Point): String = {
+    vwParamMapToString(mergeVwParams(vwParamMap, point))
+  }
+
+  def getTestVwParams(vwParamMap: Map[String, _], point: Point): String = {
+    vwParamMapToString(removeParams(mutable.Map(vwParamMap.toSeq: _*)).toMap)
+  }
+
+  def getCacheBitSize(vwParamMap: Map[String, _]): Int = {
+    vwParamMap.get("b") match {
+      case Some(b: String) => b.toInt
+      case None => vwParamMap.get("bit_precision") match {
+        case Some(b: String) => b.toInt
+        case None => 18
+        case _ => throw new IllegalArgumentException(s"Invalid bit size ${vwParamMap("bit_precision")}")
+      }
+      case _ => throw new IllegalArgumentException(s"Invalid bit size ${vwParamMap("b")}")
+    }
+  }
+
+  def parseVwArgs(args: Option[String]) = VwArgParser(args)
+}
+
+/**
+  * Parse VW parameter string and place the parameters in a Map where the key is the
+  * parameter name and the Map value is the string value for that parameter.  If no
+  * value follows the parameter name, then the Map value is an empty string.  If the
+  * parameter name is specified with a value multiple times, then the Map value is an
+  * Iterable of the multiple parameter values.
+  *
+  * This object is NOT infallible and can probably be broken, however it should work
+  * for most if not nearly all the common VW parameter strings.  Once can choose to
+  * put in parameter names that are not supported in VW.  No validation is done here to
+  * ensure the parameter is actually supported, so this should be used carefully
+  * with some common sense.  This object exists to provide convenience for VW parameter
+  * parsing and manipulation.
+  *
+  * {{{
+  *   scala> VwArgParser("-f vw.model -d dataset.txt")
+  *   res0: Map[String,String] = Map(f -> vw.model, d -> dataset.txt)
+  *
+  *   scala> VwArgParser("--loss_function logistic -c -k --passes 100")
+  *   res2: Map[String,String] = Map(loss_function -> logistic, c -> "", k -> "", passes -> 100)
+  *
+  *   scala> VwArgParser("-d train.dat --cb 4 -f cb.model  ")
+  *   res3: Map[String,String] = Map(d -> train.dat, cb -> 4, f -> cb.model)
+  *
+  *   scala> VwArgParser("--cache_file cache_train --final_regressor r_temp --passes 3 --readable_model r_te--mp.txt --l1 lambda1")
+  *   res4: Map[String,String] = Map(l1 -> lambda1, readable_model -> r_te--mp.txt, passes -> 3, cache_file -> cache_train, final_regressor -> r_temp)
+  * }}}
+  */
+object VwArgParser {
+  def oneArg(args: String): String = {
+    val spaceIndex = args.indexOf(" ")
+    if (spaceIndex != -1)
+      args.substring(0, spaceIndex)
+    else
+      args
+  }
+
+  def apply(args: String): Map[String, Any] = {
```

Comment:
To support cases where the user desires to specify `-q ab -q ac`, where the parameter name can be specified multiple times.  In these cases, the value is not a `String` but instead an `Iterable`